src/main/java/com/firebolt/jdbc/metadata/FireboltSystemEngineDatabaseMetadata.java

```diff
@@ -10,34 +9,38 @@
  * This class represents the database metadata for a system engine
  */
 public class FireboltSystemEngineDatabaseMetadata extends FireboltDatabaseMetadata {
+	private static final String INFORMATION_SCHEMA = "information_schema"; // the only schema available for system engine
 
 	public FireboltSystemEngineDatabaseMetadata(String url, FireboltConnection connection) {
 		super(url, connection);
 	}
 
 	@Override
-	public ResultSet getSchemas() {
-		return FireboltResultSet.empty();
+	public ResultSet getSchemas() throws SQLException {
+		return getSchemas(null, INFORMATION_SCHEMA);
 	}
 
 	@Override
-	public ResultSet getSchemas(String catalog, String schemaPattern) {
-		return FireboltResultSet.empty();
+	public ResultSet getSchemas(String catalog, String schemaPattern) throws SQLException {
+		return super.getSchemas(catalog, fixSchema(schemaPattern));
 	}
 
 	@Override
-	public ResultSet getColumns(String catalog, String schemaPattern, String tableNamePattern,
-			String columnNamePattern) {
-		return FireboltResultSet.empty();
+	public ResultSet getColumns(String catalog, String schemaPattern, String tableNamePattern, String columnNamePattern) throws SQLException {
+		return super.getColumns(catalog, fixSchema(schemaPattern), tableNamePattern, columnNamePattern);
 	}
 
 	@Override
-	public ResultSet getTables(String catalog, String schemaPattern, String tableNamePattern, String[] typesArr) {
-		return FireboltResultSet.empty();
+	public ResultSet getTables(String catalog, String schemaPattern, String tableNamePattern, String[] typesArr) throws SQLException {
+		return super.getTables(catalog, fixSchema(schemaPattern),  tableNamePattern, typesArr);
 	}
 
 	@Override
 	public boolean isReadOnly() throws SQLException {
 		return true;
 	}
+
+	private String fixSchema(String schemaPattern) {
+		return schemaPattern == null || INFORMATION_SCHEMA.contains(schemaPattern.replace("%", "").toLowerCase()) ? INFORMATION_SCHEMA : "does_not_exist";
+	}
```

Comment:
One issue I see here is that pattern "a%b" will not work.
Why don't we keep % in here? I believe Firebolt supports % and _ in like : https://docs.firebolt.io/sql-reference/functions-reference/like.html

Reply:
System engine can work with information_schema only. Attempt to access tables from public schema fail. This is the reason that I force to access information_schema only.

Reply:
Ok I see. And to confirm, this is when system engine is connected with no database, right? Because otherwise when it's connected to a database we can query any tables.

Reply:
As confirmed over IM we can't, I was confused. lgtm then.